src/test/java/net/snowflake/client/jdbc/PreparedStatementIT.java

```diff
@@ -0,0 +1,1721 @@
+    Date dNow = new Date(System.currentTimeMillis());
+    Date dLeapYear = new Date(951782400000L);
+    Date dFuture = new Date(32503680000000L);
+    Date[] dates = new Date[]{dEpoch, dAfterEpoch, dBeforeEpoch, dNow, dLeapYear, dFuture};
+    int[] countResult;
+
+    try
+    {
+      connection.createStatement().execute("CREATE OR REPLACE TABLE test_prepst_date (id INTEGER, d DATE)");
+      prepStatement = connection.prepareStatement("INSERT INTO test_prepst_date(id, d) VALUES(?,?)");
+
+      // First, run with non-stage binding
+      connection.createStatement().execute("ALTER SESSION SET CLIENT_STAGE_ARRAY_BINDING_THRESHOLD = 0");
+      for (int i = 0; i < dates.length; i++)
+      {
+        prepStatement.setInt(1, i);
+        prepStatement.setDate(2, dates[i]);
+        prepStatement.addBatch();
+      }
+      countResult = prepStatement.executeBatch();
+      for (int res : countResult)
+      {
+        assertEquals(1, res);
+      }
+
+      Date[] nonStageResult = new Date[dates.length];
+      ResultSet rsNonStage =
+          connection.createStatement().executeQuery("SELECT * FROM test_prepst_date ORDER BY id ASC");
+      for (int i = 0; i < nonStageResult.length; i++)
+      {
+        rsNonStage.next();
+        nonStageResult[i] = rsNonStage.getDate(2);
+      }
+
+      connection.createStatement().execute("DELETE FROM test_prepst_date WHERE 1=1");
+
+      // Now, run with stage binding
+      connection.createStatement().execute("ALTER SESSION SET CLIENT_STAGE_ARRAY_BINDING_THRESHOLD = 1"); // enable stage bind
+      for (int i = 0; i < dates.length; i++)
+      {
+        prepStatement.setInt(1, i);
+        prepStatement.setDate(2, dates[i]);
+        prepStatement.addBatch();
+      }
+      countResult = prepStatement.executeBatch();
+      for (int res : countResult)
+      {
+        assertEquals(1, res);
+      }
+
+      Date[] stageResult = new Date[dates.length];
+      ResultSet rsStage = connection.createStatement().executeQuery("SELECT * FROM test_prepst_date ORDER BY id ASC");
+      for (int i = 0; i < stageResult.length; i++)
+      {
+        rsStage.next();
+        stageResult[i] = rsStage.getDate(2);
+      }
+
+      for (int i = 0; i < dates.length; i++)
+      {
+        assertEquals("Stage binding date should match non-stage binding date",
+                     nonStageResult[i], stageResult[i]);
+      }
+
+    }
+    finally
+    {
+      connection.createStatement().execute("DROP TABLE IF EXISTS test_prepst_date");
+    }
+  }
+
+
+  @Test
+  public void testClearParameters() throws SQLException
+  {
+    connection = getConnection();
+    prepStatement = connection.prepareStatement(insertSQL);
+    bindOneParamSet(prepStatement, 1, 1.22222, (float) 1.2, "test", 12121212121L, (short) 12);
+    prepStatement.clearParameters();
+    bindOneParamSet(prepStatement, 3, 1.22, 1.2f, "hello", 12222L, (short) 1);
+    prepStatement.addBatch();
+    prepStatement.executeBatch();
```

Comment:
can the test verify executeBatch cleans up all internal data? We know it is today, but not sure future. Can you find out how to detect memory leak here?